src/js/oncoprintmodel.ts

```diff
@@ -1153,13 +1156,21 @@ export default class OncoprintModel {
             this.track_expand_button_getter[track_id] = params.expandButtonTextGetter;
         }
 
+        this.track_sort_direction[track_id] = ifndef(params.init_sort_direction, 1);
+
         this.track_can_show_gaps[track_id] = ifndef(params.track_can_show_gaps, false);
-        this.track_show_gaps[track_id] = ifndef(params.show_gaps_on_init, false);
+        const trackShowGaps = ifndef(params.show_gaps_on_init, false);
+        this.track_show_gaps[track_id] = trackShowGaps;
+        const trackNotSorted = this.track_sort_direction[track_id] === 0;
+        if (trackShowGaps && trackNotSorted) {
+            this.track_sort_direction[track_id] = 1;
```

Comment:
I don't understand the logic for this. Why does showing gaps mean you should override the default sorting?

Reply:
Gaps are a bit of space between the different groups or categories in a track. In order to show gaps between the categories, we need to sort that track first.